pilot/pkg/config/kube/gateway/testdata/tls.yaml

```diff
@@ -48,6 +48,19 @@ spec:
       - name: my-cert-http
       options:
         gateway.istio.io/tls-terminate-mode: MUTUAL
+  - name: terminate-istio-mtls
+    hostname: "egress.example"
+    port: 34000
+    protocol: HTTPS
+    allowedRoutes:
+      namespaces:
+        from: All
+    tls:
+      mode: Terminate
+      certificateRefs:
+      - name: dummy # TODO: revmove this when https://github.com/kubernetes-sigs/gateway-api/pull/2721 is in
```

Comment:
should we just wait for this to land?

Reply:
I could go either way. My thought is that we will wait to use it anywhere (e.g., no docs with this hack), but that it would be nice that the implementation is there as soon as the CRDs are fixed, no code change needed, just a cleanup in the unit test. Anyway, your call if you want to hold this for now.

Reply:
Not sure i understand, this cert if used to terminate tls from outside client. Why `ISTIO_MUTUAL` does need this cert?

Reply:
This is used for in-mesh clients calling an egress gateway.

Reply:
is there a xRoute, and what kind of route is allowed to bind with this gateway

Reply:
Istio-specific extensions allow you to bind an xRoute to a `ServiceEntry` to direct traffic to the egress gateway, then bind an xRoute to the gateway to direct traffic to an external `Hostname`. See https://preliminary.istio.io/latest/docs/tasks/traffic-management/egress/egress-gateway/#egress-gateway-for-http-traffic